manticore/ethereum.py

```diff
@@ -389,17 +389,25 @@ def simplify(x):
             value = arithmetic_simplifier(Operators.CONCAT(size, *map(Operators.ORD, data[offset + padding:offset + padding + byte_size])))
             return simplify(value)
 
-        if ty == u'uint256':
-            return get_uint(256, offset), offset + 32
-        elif ty in (u'bool', u'uint8'):
+        if ty.startswith('uint') and 0 <= int(ty[4:]) <= 256:
```

Comment:
i think we should move this `0 <= ... <= 256` range check inside the `if` body and add it to the check where we raise an EthereumError. if we are given something like `uint3000` that is an invalid uint size, but we would raise a `NotImplementedError`

Reply:
same below with int